apps/gk_field_multib.c

```diff
@@ -0,0 +1,229 @@
+#include <gkyl_alloc.h>
```

Comment:
you have huge lines in this file, please shorten

also, maybe we should call this gk_multib_field, that way if we create more gk_ objects for MB, they'll be grouped together in apps/ (also for the app we placed multib after gyrokinetic)